src/librustc_typeck/check/op.rs

```diff
@@ -254,6 +256,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 if !lhs_ty.references_error() && !rhs_ty.references_error() {
```

Comment:
Pre-existing, but couldn't we make this into a match arm to avoid indenting the next 300 lines?

```rust
Err(()) if lhs_ty.references_error() || rhs_ty.references_error() => self.tcx.ty_error(),
Err(()) => {
 // as before..
}
```

Reply:
Done